Sources/Basic/StringExtensions.swift

```diff
@@ -0,0 +1,32 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+extension Array where Element == String {
+    /// Invokes the given closure on the contents of the string array, represented as a pointer
+    /// to a null-terminated sequence of pointers to null-terminated sequences of UTF-8 code units.
+    ///
+    /// The `withCStringArray(_:)` method ensures that the sequence's lifetime extends through
+    /// the execution of body. The pointer argument to body is only valid for the lifetime of
+    /// the closure. Do not escape it from the closure for later use.
+    ///
+    /// - Parameters:
+    ///     - body: A closure that takes a pointer to the array's sequence of C-Strings as its
+    ///       sole argument. If the closure has a return value, it is used as the return value
+    ///       of the `withCStringArray(_:)` method. The pointer argument is valid only for the
+    ///       duration of the closure's execution.
+    /// - Returns: The return value of the body closure, if any.
+    public func withCStringArray<Result>(_ body: ([UnsafeMutablePointer<Int8>?]) throws -> Result) rethrows -> Result {
```

Comment:
I think we should create a new type, `CStrings` or something, and then deallocate in `deinit`.